### PR TITLE
[FEATURE] dashboard tabs with iframes functionality (Podio feature_89)

### PR DESCRIPTION
**Pull request by yosokus**

### CODE REVIEW

File: Classes/DashboardWidgets/IframeWidget.php [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+ *                                                                        */
+
+use TYPO3\CMS\Dashboard\DashboardWidgetInterface;
+use TYPO3\CMS\Dashboard\Domain\Model\DashboardWidgetSettings;
+use TYPO3\CMS\Fluid\View\StandaloneView;
+
+class IframeWidget extends AbstractWidget implements DashboardWidgetInterface
+{
+
+    const IDENTIFIER = 1487642496;
```
Comment (MattiasNilsson) [inline Classes/DashboardWidgets/IframeWidget.php:24]: Identifier for what?

File: Classes/DashboardWidgets/IframeWidget.php [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+     * @return string
+     */
+    private function generateContent()
+    {
+        $widgetTemplateName = $this->widget['template'];
+        /** @var StandaloneView $actionView */
+        $actionView = \TYPO3\CMS\Core\Utility\GeneralUtility::makeInstance('TYPO3\\CMS\\Extbase\\Object\\ObjectManager')->get('TYPO3\\CMS\\Fluid\\View\\StandaloneView');
+        $template = \TYPO3\CMS\Core\Utility\GeneralUtility::getFileAbsFileName($widgetTemplateName);
+        $actionView->setTemplatePathAndFilename($template);
+        $actionView->assign('url', $this->url);
```
Comment (MattiasNilsson) [inline Classes/DashboardWidgets/IframeWidget.php]: replace usage of assign with assignMultiple

**Comment by MattiasNilsson** [comment]
Looks good, just some minor things.

**Review by MattiasNilsson — CHANGES_REQUESTED** [review]

File: Configuration/TCA/DashboardWidgetSettings.php [hunk 3]
```diff
@@ -109,9 +109,11 @@
     ),
 );
 ## EXTENSION BUILDER DEFAULTS END TOKEN - Everything BEFORE this line is overwritten with the defaults of the extension builder
+$iframeWidgetId = \TYPO3\CMS\Dashboard\DashboardWidgets\IframeWidget::IDENTIFIER;
```
Comment (yosokus) [inline Configuration/TCA/DashboardWidgetSettings.php:112]: @MattiasNilsson the identifier is used here

File: ext_tables.php [hunk 4]
```diff
@@ -106,6 +106,14 @@
             'class' => 'TYPO3\\CMS\\Dashboard\\DashboardWidgets\\SysNewsWidget',
             'template' => 'EXT:dashboard/Resources/Private/Templates/DashboardWidgets/SysNewsWidget.html',
             'size' => '1x2',
+        ),
+        \TYPO3\CMS\Dashboard\DashboardWidgets\IframeWidget::IDENTIFIER => array(
```
Comment (yosokus) [inline ext_tables.php:110]: @MattiasNilsson the identifier is also used here

File: Classes/DashboardWidgets/IframeWidget.php (hunk 1 above)
Comment (yosokus) [reply Classes/DashboardWidgets/IframeWidget.php:24]: @MattiasNilsson that the widget identifier. It is used in 2 files so I used a constant for easy retrieval.

File: Resources/Private/Templates/DashboardWidgets/IframeWidget.html [hunk 5]
```diff
@@ -0,0 +1,3 @@
+<f:if condition="{url}">
+  <iframe class="iframe-widget" src="{url}" width="100%" height="100%" scrolling="{scrolling}"></iframe>
```
Comment (yosokus) [inline Resources/Private/Templates/DashboardWidgets/IframeWidget.html:2]: @MattiasNilsson  should I remove the scrolling option? I'm not sure if it is needed.

Comment (MattiasNilsson) [reply Resources/Private/Templates/DashboardWidgets/IframeWidget.html:2]: Keep it for now.

**Review by MattiasNilsson — APPROVED** [review]

**Pull request merged**